File: backend/src/main/java/com/svelteup/app/backend/profile/dtos/SvelteUpUserProfileDto.java
package com.svelteup.app.backend.profile.dtos;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.svelteup.app.backend.productorder.dto.ProductReviewScoreCardDto.PutProductReviewScoreCardDto;
import com.svelteup.app.backend.profile.models.SvelteUpUserProfile;

import java.util.ArrayList;
import java.util.List;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class SvelteUpUserProfileDto {
   public Integer phoneNumberAreaCode;
   public Integer phoneNumberPhoneNumber;

   public String firstName;
   public String lastName;
   public String email;

   public String addressLineOne;
   public String userProfileCity;
   public String userProfileState;
   public String userProfileCountry;
   public Integer zipCode;

   public String aboutMe;
   public String image;

   public PutProductReviewScoreCardDto productReviewScoreCard;

   List<String> foeList;

   public SvelteUpUserProfileDto(SvelteUpUserProfile profileEntity,String awsImage)
   {
      this.phoneNumberAreaCode = profileEntity.getPhoneNumberAreaCode();
      this.phoneNumberPhoneNumber = profileEntity.getPhoneNumberPhoneNumber();
      this.firstName = profileEntity.getFirstName();
      this.lastName = profileEntity.getLastName();
      this.email = profileEntity.getEmail();
      this.addressLineOne = profileEntity.getAddressLineOne();
      this.userProfileCity = profileEntity.getUserProfileCity();
      this.userProfileState = profileEntity.getUserProfileState();
      this.userProfileCountry = profileEntity.getUserProfileCountry();
      this.zipCode = profileEntity.getUserProfileZipCode();
      this.aboutMe =  profileEntity.getAboutMe();
      this.image = awsImage;
      this.productReviewScoreCard = profileEntity.getProductReviewScoreCard() != null ? profileEntity.getProductReviewScoreCard().toExistingDto():null;
      foeList = new ArrayList<>();
   }
}

File: frontend/src/Dto/profile/SvelteUpUserProfileDto.ts
import type {ProductReviewScoreCardDto} from "./ProductReviewScoreCardDto";

class SvelteUpUserProfileDto
{
    public  phoneNumberAreaCode:number|null;
    public  phoneNumberPhoneNumber:number|null

    public  firstName:string|null;
    public  lastName:string|null;
    public  email:string|null;

    public addressLineOne:string|null;
    public userProfileCity:string|null;
    public userProfileState:string|null;
    public userProfileCountry:string|null;
    public zipCode:number|null;

    public aboutMe:string|null;
    public image:string|null;
    public productReviewScoreCard: ProductReviewScoreCardDto;

    public constructor() {}

    public buildEmailDto(email:string)
    {
        this.email = email;
    }

    public buildProfileDto(image:any, aboutMe:string)
    {
        this.image = image;
        this.aboutMe = aboutMe;
    }

    public buildIdentityDto(firstName:string, lastName:string, areaCode:number,phoneNumberNumber:number, addressLine:string, userProfileState:string, userProfileCountry:string)
    {
        this.firstName  = firstName;
        this.lastName = lastName;
        this.phoneNumberAreaCode = areaCode;
        this.phoneNumberPhoneNumber = phoneNumberNumber;
        this.addressLineOne  = addressLine;
        this.userProfileState  = userProfileState;
        this.userProfileCountry  = userProfileCountry;
    }


}

export default SvelteUpUserProfileDto;